File: java/src/main/java/com/mandiri/qris/entity/MandiriQrisPayment.java
package com.mandiri.qris.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "mandiri_qris_payments")
public class MandiriQrisPayment {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(unique = true, nullable = false)
    private String qrId;
    
    @Column(unique = true, nullable = false)
    private String reference;
    
    @Column(columnDefinition = "TEXT", nullable = false)
    private String qrString;
    
    @Column(length = 500, nullable = false)
    private String qrImageUrl;
    
    @Column(nullable = false)
    private Double amount;
    
    @Column(nullable = false)
    private String status = "PENDING";
    
    @Column
    private String transactionId;
    
    @Column
    private LocalDateTime paidAt;
    
    @Column(nullable = false)
    private LocalDateTime expiredAt;
    
    @Column(columnDefinition = "TEXT")
    private String metadata;
    
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
    
    @Column
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: aspnet/Models/MandiriQrisPayment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MandiriQris.Models
{
    [Table("mandiri_qris_payments")]
    public class MandiriQrisPayment
    {
        [Key]
        public long Id { get; set; }

        [Required]
        [MaxLength(255)]
        public string QrId { get; set; } = string.Empty;

        [Required]
        [MaxLength(255)]
        public string Reference { get; set; } = string.Empty;

        [Required]
        [Column(TypeName = "TEXT")]
        public string QrString { get; set; } = string.Empty;

        [Required]
        [MaxLength(500)]
        public string QrImageUrl { get; set; } = string.Empty;

        [Required]
        public double Amount { get; set; }

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "PENDING";

        [MaxLength(255)]
        public string? TransactionId { get; set; }

        public DateTime? PaidAt { get; set; }

        [Required]
        public DateTime ExpiredAt { get; set; }

        [Column(TypeName = "TEXT")]
        public string? Metadata { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }
    }
}
